fileio/ft_read_data.m

```diff
@@ -1372,6 +1372,21 @@
       dat = read_yokogawa_data(filename, hdr, begsample, endsample, chanindx);
     end
     
+  case 'blackrock_nsx'
+    % use the NPMK toolbox for the file reading
+    ft_hastoolbox('NPMK', 1);
+    
+    % ensure that the filename contains a full path specification,
+    % otherwise the low-level function fails
+    [p,f,e] = fileparts(filename);
+    if ~isempty(p)
+      % this is OK
+    elseif isempty(p)
+      filename = which(filename);
+    end
+    orig = openNSx(filename, 'duration', [begsample endsample], 'channels', chanindx);
+    dat  = double(orig.Data);
```

Comment:
we could replace line 1387/1388 with a conditional statement. This would require that it is known a priori, whether the datafile consists of a single continuous piece of data, or of multiple small pieces of (discontinuous) stuff. I guess that begsample/endsample handling would need to be carefully implemented and checked in that case as well. How easy would it be for you to create a small example dataset?